src/components/UtilMenu/index.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { withRouter } from 'react-router-dom'
+
+import { Icon } from '@codespec/react-toolset'
+import IconButton from 'material-ui/IconButton'
+import IconMenu from 'material-ui/IconMenu'
+import MenuItem from 'material-ui/MenuItem'
+
+export class UtilMenu extends Component {
+  static propTypes = {
+    isLogged: PropTypes.bool,
+    onClickSign: PropTypes.func
+  }
+
+  static defaultProps = {
+    isLogged: false
+  }
+
+  routeToPage = (page) => {
+    this.props.history.push(page)
+  }
+
+  render() {
+    const { isLogged, onClickSign } = this.props
+
+    return (
+      <IconMenu
+        iconButtonElement={
+          <IconButton>
+            <Icon type='materialDesign' name='MdMoreVert' size={26} color="#fff" />
+          </IconButton>
+        }
+        targetOrigin={{ horizontal: 'right', vertical: 'top' }}
+        anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
+        touchTapCloseDelay={10}
+      >
+        { isLogged && <MenuItem primaryText='Setting' />}
+        { isLogged && <MenuItem primaryText='Help' />}
+        { isLogged && <MenuItem
+          primaryText='Sign Out'
+          onClick={onClickSign} /> }
+        { !isLogged && <MenuItem
+          primaryText='Sign In'
+          onClick={onClickSign} /> }
+        { !isLogged && <MenuItem
+          primaryText='Register'
+          onClick={() => this.routeToPage('register')} />
+        }
```

Comment:
I cannot do like this
```jsx
{ isLogged
? <div>
  <MenuItem />
  <MenuItem />
  <MenuItem />
</div>
: <div>
  <MenuItem />
  <MenuItem />
</div>
}
```
Due to the limitation of IconMenu of Material-ui, the MenuItem is wrapped with any other html element, it is not closed when the menu is clicked.

https://github.com/callemall/material-ui/issues/7119